www/js/goals.js

```diff
@@ -135,43 +151,55 @@ angular.module('emission.main.goals',['emission.services', 'emission.plugin.logg
     /*$scope.data = {
         showDelete: false
     };*/
-    $scope.toBrowser = function() {
-            var options = {
-              location: 'no',
-              clearcache: 'no',
-              toolbar: 'yes'
-            };
-
-            var settings = localStorage.getItem("habit-mobile-settings");
-
-          $rootScope.$on('$cordovaInAppBrowser:loadstop', function(e, event){
-            // insert Javascript via code / file
-            if (event.url == 'https://em-game.eecs.berkeley.edu/static/front' ||
-                event.url == 'https://em-game.eecs.berkeley.edu/static/front#/tasks') {
-                $cordovaInAppBrowser.executeScript({
-                  code: "localStorage.setItem('habit-mobile-settings', '" + settings + "');" 
-                  + "window.location.href = 'https://em-game.eecs.berkeley.edu/#/tasks';"
-                });
-            } else {
-                Logger.log("checking for game loadstop, finished loading url "+event.url+" ignoring...");
-            }
-          });
-        $cordovaInAppBrowser.open('https://em-game.eecs.berkeley.edu/#/tasks', '_blank', options)
-          .then(function(event) {
-            // success
-          })
-          .catch(function(event) {
-            // error
+    $scope.toBrowser = function() {     
+            var options = {       
+              location: 'no',     
+              clearcache: 'no',       
+              toolbar: 'yes'      
+            };        
+      
+            var settings = localStorage.getItem("habit-mobile-settings");     
+      
+          $rootScope.$on('$cordovaInAppBrowser:loadstop', function(e, event){     
+            // insert Javascript via code / file      
+            if (event.url == 'https://em-game.eecs.berkeley.edu/static/front' ||      
+                event.url == 'https://em-game.eecs.berkeley.edu/static/front#/tasks') {       
+                $cordovaInAppBrowser.executeScript({      
+                  code: "localStorage.setItem('habit-mobile-settings', '" + settings + "');"      
+                  + "window.location.href = 'https://em-game.eecs.berkeley.edu/#/tasks';"     
+                });       
+            } else {      
+                Logger.log("checking for game loadstop, finished loading url "+event.url+" ignoring...");     
+            }     
+          });     
+        $cordovaInAppBrowser.open('https://em-game.eecs.berkeley.edu/#/tasks', '_blank', options)     
+          .then(function(event) {     
+            // success        
+          })      
+          .catch(function(event) {        
+            // error      
           });
-
+       
     };
+
     $scope.openLeaderboard = function() {
         $scope.leaderboardModal.show();
     };
     $scope.closeLeaderboard = function() {
         $scope.leaderboardModal.hide();
     };
-
+    $scope.openChallenge = function() {
+        $scope.challengeModal.show();
```

Comment:
we should be able to restructure this to a separate service later.
check out https://github.com/e-mission/e-mission-data-collection/tree/master/www/ui/ionic
for an example